OasisDEX/oasis-borrow
# Infura calls: Only request data for cards we show

# [Only request data for cards we show](https://app.shortcut.com/oazo-apps/story/5171/don-t-request-all-product-card-data-when-only-showing-a-few-product-cards)

Infura calls reduction:

- Load homepage: 324 to about 40 calls.
- Navigate around website for a 10-block user-session: 
  - before (10 blocks duration): 4000 calls
  - after (7 blocks duration): 347 calls

## Changes 👷‍♀️
- Only request data for product cards that are displayed.   Home page, assets pages, product pages.
  
## How to test 🧪
  <Please explain how to test your changes>
- All cards display across all pages.  Verify button action.
- Open/close a few positions.

Some images
-----

Before:

<img width="512" alt="Pasted Graphic 7" src="https://user-images.githubusercontent.com/3783198/180421520-115ea352-252d-4b55-b49a-f39270789e31.png">

After:

<img width="472" alt="Pasted Graphic 6" src="https://user-images.githubusercontent.com/3783198/180422124-51f323f8-a20e-4ff5-bb08-53028f5f525c.png">
